inspirehep/modules/migrator/tasks.py

```diff
@@ -154,6 +159,52 @@ def migrate_chunk(chunk, broken_output=None, dry_run=False):
         models_committed.connect(record_modification)
 
 
+@celery.task(ignore_result=True)
+def add_citation_counts():
+    from elasticsearch.helpers import bulk as es_bulk
+    from elasticsearch.helpers import scan as es_scan
+    from collections import Counter
+
+    # list of update-jsons for each record
+    records_to_update = []
+
+    # lookup dictionary where key: recid of the record
+    # and value: number of records that cite that record
+    citations_lookup = Counter()
+    for record in es_scan(
+            es,
+            query={
+                "query": {
+                    "match_all": {}},
+                "size": CHUNK_SIZE},
+            scroll=u'1m',
+            index="hep",
+            doc_type="record"):
+
+        # update lookup dictionary based on references of the record
+        if 'references' in record['_source']:
+            references = record['_source']['references']
+            for reference in references:
+                recid = unicode(reference.get('recid'))
+                if recid is not None:
+                    citations_lookup[recid] += 1
+
+        # prepare json to update the record
+        json = {'_op_type': 'update',
+                '_index': 'hep',
+                '_type': 'record',
+                '_id': record['_id'],
+                }
+        records_to_update.append(json)
+
+    for json in records_to_update:
+        citation_count = citations_lookup[json['_id']]
+        json.update({'doc': {'citation_count': citation_count}})
```

Comment:
I believe it's correct - https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html#_updates_with_a_partial_document